src/codeu/chat/server/Controller.java

```diff
@@ -143,29 +148,113 @@ public ConversationHeader newConversation(Uuid id, String title, Uuid owner, Tim
     return conversation;
   }
 
+  public Interest addInterest(Uuid userId, Uuid interestId, Type interestType) {
+    return addInterest(userId, userId, interestId, interestType, Time.now());
+  }
+
+  public Interest addInterest(Uuid id, Uuid userId, Uuid interestId,
+      Type interestType, Time creationTime) {
+    return model.addInterest(id, userId, interestId, interestType, creationTime);
+  }
+
+  public void removeInterest(Uuid userId, Uuid interestId) {
+    model.removeInterest(userId, interestId);
+  }
+
+  // TODO(Adam): Create and send an ArrayList of InterestStatus objects instead of an array
+  // list of strings. Allow the client to figure out how to sort through the
+  // data.
+  public ArrayList<InterestStatus> interestStatus(Uuid user) {
+    ArrayList<Uuid> userInterests = model.interests.get(user);
+    ArrayList<InterestStatus> result = new ArrayList<>();
+    Time now = Time.now();
+    if (userInterests == null) return result;
+    for (Uuid interestId : userInterests) {
+      InterestStatus report = processInterest(interestId, now);
+      if (report == null) continue;
+      result.add(report);
+    }
+    return result;
+  }
+
+  /* Syntax guide: 
+   *
+   * if type is User:
+   *   U <name>: C <header title> --> for conversations created by the user
+   *   U <name>: A <header title> --> for messages added to conversation by the
+   *   user.
+   * if type is Conversation
+   *   C <name>: <number of new messages>
+   */
+  private InterestStatus processInterest(Uuid id, Time now) {
+    Interest interest = model.interestById().first(id);
+    if (interest == null) return null;
+    Time lastUpdate = interest.lastUpdate;
+    InterestStatus result = null;
+    if (interest.type == Type.USER) {
+      User user = model.userById().first(id);
+      // Return all values with time higher than last Update
+      Iterable<ConversationHeader> headers = model.conversationByTime().
+        after(lastUpdate);
+      List<String> createdConversations = new ArrayList<>();
+      for (ConversationHeader header : headers) {
+        if (!header.owner.equals(user.id)) continue;
```

Comment:
Don't need `continue.` Just don't negate the condition, and put the next line inside the `if` block.